esmf_regrid/tests/unit/experimental/unstructured_scheme/test_regrid_unstructured_to_unstructured.py

```diff
@@ -0,0 +1,220 @@
+    with pytest.raises(NotImplementedError):
+        _ = regrid_unstructured_to_unstructured(face_src, tgt, method="other")
+    with pytest.raises(ValueError) as excinfo:
+        _ = regrid_unstructured_to_unstructured(node_src, tgt, method="conservative")
+    expected_message = (
+        "Conservative regridding requires a source cube located on "
+        "the face of a cube, target cube had the node location."
+    )
+    assert expected_message in str(excinfo.value)
+    with pytest.raises(ValueError) as excinfo:
+        _ = regrid_unstructured_to_unstructured(edge_src, tgt, method="bilinear")
+    expected_message = (
+        "bilinear regridding requires a source cube with a node "
+        "or face location, target cube had the edge location."
+    )
+    assert expected_message in str(excinfo.value)
+    with pytest.raises(ValueError) as excinfo:
+        _ = regrid_unstructured_to_unstructured(edge_src, tgt, method="nearest")
+    expected_message = (
+        "nearest regridding requires a source cube with a node "
+        "or face location, target cube had the edge location."
+    )
+    assert expected_message in str(excinfo.value)
+
+
+def test_multidim_cubes():
+    """
+    Test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_unstructured_to_unstructured`.
+
+    Tests with multidimensional cubes. The source cube contains
+    coordinates on the dimensions before and after the mesh dimension.
+    """
+    tgt = _flat_mesh_cube()
+    tgt_mesh = tgt.mesh
+    mesh = _full_mesh()
+    mesh_length = mesh.connectivity(contains_face=True).shape[0]
+    tgt_mesh_length = mesh.connectivity(contains_face=True).shape[0]
+
+    h = 2
+    t = 3
+    height = DimCoord(np.arange(h), standard_name="height")
+    time = DimCoord(np.arange(t), standard_name="time")
+
+    src_data = np.empty([t, mesh_length, h])
+    src_data[:] = np.arange(t * h).reshape([t, h])[:, np.newaxis, :]
+    cube = Cube(src_data)
+    mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+    cube.add_aux_coord(mesh_coord_x, 1)
+    cube.add_aux_coord(mesh_coord_y, 1)
+    cube.add_dim_coord(time, 0)
+    cube.add_dim_coord(height, 2)
+
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    tgt = _gridlike_mesh_cube(n_lons, n_lats)
+
+    result = regrid_unstructured_to_unstructured(cube, tgt)
+
+    # Lenient check for data.
+    expected_data = np.empty([t, n_lats, n_lons, h])
+    expected_data[:] = np.arange(t * h).reshape(t, h)[:, np.newaxis, np.newaxis, :]
+    assert np.allclose(expected_data, result.data)
+
+    expected_cube = Cube(expected_data)
+    expected_cube.add_dim_coord(time, 0)
+    expected_cube.add_dim_coord(tgt.coord("latitude"), 1)
+    expected_cube.add_dim_coord(tgt.coord("longitude"), 2)
+    expected_cube.add_dim_coord(height, 3)
+
+    # Check metadata and scalar coords.
+    result.data = expected_data
+    assert expected_cube == result
+
+
+def test_resolution():
```

Comment:
This test can be removed since resolution isn't needed for mesh to mesh regridding.